docs/features.md

```diff
@@ -11,33 +11,220 @@ The first generic type is the getter return type, and the second is the setter a
 #### Examples
 
 ```php
+<?php
 /** @return Attribute<string[], string[]> */
 protected function scopes(): Attribute
 {
-	return Attribute::make(
-		get: fn (?string $value) => is_null($value) ? [] : explode(' ', $value),
-		set: function(array $value) {
-			$set = array_unique($value);
-			sort($set);
-			return ['scopes' => implode(' ', $set)];
-		}
-	);
+    return Attribute::make(
+        get: fn (?string $value) => is_null($value) ? [] : explode(' ', $value),
+        set: function(array $value) {
+            $set = array_unique($value);
+            sort($set);
+            return ['scopes' => implode(' ', $set)];
+        }
+    );
 }
 ```
 
 ```php
+<?php
 /** @return Attribute<bool, never> */
 protected function isTrue(): Attribute
 {
-	return Attribute::make(
-		get: fn (?string $value): bool => $value === null,
-	);
+    return Attribute::make(
+        get: fn (?string $value): bool => $value === null,
+    );
 }
 ```
 
+## Custom Model Builders
+
+Custom builders offer a better static analysis experience than using model scopes, and they help slim down the model class.
+
+Here's an example of how to create a custom builder class:
+
+```php
+<?php
+
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\HasBuilder;
+use Illuminate\Database\Eloquent\Model;
+
+/** @extends Builder<User> */
+class UserBuilder extends Builder
+{
+    /** @return $this */
+    public function active(): static
+    {
+        return $this->where('active', true);
```

Comment:
This will return `static(UserBuilder)`.

Adding the where and then returning `$this` will instead return `$this(UserBuilder)` as expected.
```suggestion
        $this->where('active', true);

        return $this;
```